### PR TITLE
Escape hatch cancel orders

### PR DESCRIPTION
**Pull request by nuevoalex**
[Clubhouse Story](https://app.clubhouse.io/augur/story/10734)

## Submitter checklist
- [ ] Test covering functionality added/fixed in
- Check the linked story includes the following:
  - [ ] Steps to reproduce
  - [ ] Screenshot (If applicable)

## Reviewer Checklist
- [ ] Test/lint pass

### CODE REVIEW

**Comment by nuevoalex** [comment]
To test this you'll need to either use this branch of AN or wait till it gets merged:

https://github.com/AugurProject/augur-node/pull/361

File: src/modules/escape-hatch/actions/withdraw-funds-in-emergency.js [hunk 1]
```diff
@@ -85,5 +87,33 @@ export default function (ownedMarkets, marketsWithShares, callback = logError) {
...
+          dispatch(updateOrderClearEscrowed(orderId))
+          if (orderHasSharesEscrowed) {
+            dispatch(doUpdateShareFrozenValue(order.marketId, dispatch, () => {
+              augur.api.TradingEscapeHatch.claimSharesInUpdate({
+                meta: loginAccount.meta,
+                _market: order.marketId,
+                onSent: noop,
+                onSuccess: (res) => {
+                  console.log('TradingEscapeHatch.claimSharesInUpdate', res)
+                  dispatch(updateMarketFrozenSharesValue(order.marketId, 0))
```
Comment (bthaile) [inline src/modules/escape-hatch/actions/withdraw-funds-in-emergency.js]: probably don't need this console log.

File: src/modules/escape-hatch/selectors/escape-hatch-data.js [hunk 2]
```diff
@@ -71,7 +76,14 @@ export const getEscapeHatchData = createSelector(
...
 
-    data.fundsAvailableForWithdrawal = data.rep + data.eth > 0
+    Object.keys(orders).forEach((orderId) => {
+      const order = orders[orderId]
+      data.eth = data.eth.plus(createBigNumber(speedomatic.fix(order.tokensEscrowed)))
+      data.shares = data.shares.plus(createBigNumber(speedomatic.fix(order.sharesEscrowed)))
+      data.gas = data.gas.plus(CANCEL_ORDER_GAS_ESTIMATE)
+    })
+
+    data.fundsAvailableForWithdrawal = data.rep + data.eth + data.shares > 0
```
Comment (bthaile) [inline src/modules/escape-hatch/selectors/escape-hatch-data.js]: is '+' same as .plus with BigNumbers?

`fundsAvailableForWithdrawal` seems to be used as a number but is a bool. 

referencing src/modules/escape-hatch/components/escape-hatch.jsx
`  const fundsAvailableForWithdrawal = nextProps.escapeHatchData.fundsAvailableForWithdrawal > 0`

How about using it as a bool or use it as a number. If use as number use .plus(....).plus(...).toNumber()

**Comment by bthaile** [comment]
Created an order for 100 shares at .1, the `get-market-balance` script says I have 100 shares.

After pulling escape hatch, I expected to see Shares of 100.

Withdrawing funds gives the correct amount of ETH. Might be I'm missing the correct trading setup.

screen shot 2018-05-17 at 9 45 40 am


screen shot 2018-05-17 at 9 46 03 am

File: src/modules/escape-hatch/selectors/escape-hatch-data.js (hunk 2 above)
Comment (nuevoalex) [reply src/modules/escape-hatch/selectors/escape-hatch-data.js]: ah yea good point, will do

**Comment by bthaile** [comment]
`const fundsAvailableForWithdrawal = nextProps.escapeHatchData.fundsAvailableForWithdrawal > 0`
does this need to be fundsAvailableForWithdrawal.gt(ZERO) // ZERO using constant

`import { ZERO } from 'modules/trade/constants/numbers'`
just for convention.

**Pull request merged**